src/bp/core/routers/bots/nlu.ts

```diff
@@ -0,0 +1,305 @@
+      })
+    )
+
+    this.router.get(
+      '/intents/:intent',
+      this._checkTokenHeader,
+      this._needPermissions('read', 'bot.content'),
+      this.asyncMiddleware(async (req, res) => {
+        const { botId, intent } = req.params
+        const intentDef = await this.nluService.intents.getIntent(botId, intent)
+        res.send(intentDef)
+      })
+    )
+
+    this.router.post(
+      '/intents/:intent/delete',
+      this._checkTokenHeader,
+      this._needPermissions('write', 'bot.content'),
+      this.asyncMiddleware(async (req, res) => {
+        const { botId, intent } = req.params
+        try {
+          await this.nluService.intents.deleteIntent(botId, intent)
+          res.sendStatus(204)
+        } catch (err) {
+          this.logger
+            .forBot(botId)
+            .attachError(err)
+            .error('Could not delete intent')
+          res.status(400).send(err.message)
+        }
+      })
+    )
+
+    this.router.post(
+      '/intents',
+      this._checkTokenHeader,
+      this._needPermissions('write', 'bot.content'),
+      this.asyncMiddleware(async (req, res) => {
+        const { botId } = req.params
+        try {
+          const intentDef = await validate(req.body, IntentDefCreateSchema, {
+            stripUnknown: true
+          })
+
+          await this.nluService.intents.saveIntent(botId, intentDef)
+
+          res.sendStatus(200)
+        } catch (err) {
+          this.logger
+            .forBot(botId)
+            .attachError(err)
+            .warn('Cannot create intent')
+          res.status(400).send(err.message)
+        }
+      })
+    )
+
+    this.router.post(
+      '/intents/:intentName',
+      this._checkTokenHeader,
+      this._needPermissions('write', 'bot.content'),
+      this.asyncMiddleware(async (req, res) => {
+        const { botId, intentName } = req.params
+        try {
+          await this.nluService.intents.updateIntent(botId, intentName, req.body)
+          res.sendStatus(200)
+        } catch (err) {
+          this.logger
+            .forBot(botId)
+            .attachError(err)
+            .error('Could not update intent')
+          res.sendStatus(400)
+        }
+      })
+    )
+
+    this.router.post(
+      '/condition/intentChanged',
+      this._checkTokenHeader,
+      this._needPermissions('write', 'bot.content'),
+      this.asyncMiddleware(async (req, res) => {
+        const { botId } = req.params
+        const { action } = req.body
+        const condition = req.body.condition as sdk.DecisionTriggerCondition
+
+        if (action === 'delete' || action === 'create') {
+          try {
+            await this.nluService.intents.updateContextsFromTopics(botId, [condition!.params!.intentName])
+            return res.sendStatus(200)
+          } catch (err) {
+            return res.status(400).send(err.message)
+          }
+        }
+
+        res.sendStatus(200)
+      })
+    )
+
+    this.router.post(
```

Comment:
we should remove all notion of topic in this pr

Reply:
tbh I don't even know what this is... I'll take a look before deleting